packages/react-core/src/components/Chip/__tests__/Chip.test.tsx

```diff
@@ -1,59 +1,234 @@
+  const isOverflowChip = chipType === 'overflow chip';
+
+  test(`Renders without children for ${chipType}`, () => {
+    render(
+      <div data-testid="container">
+        <Chip isOverflowChip={isOverflowChip} />
+      </div>
+    );
+
+    expect(screen.getByTestId('container').firstChild).toBeVisible();
+  });
+
+  test(`Renders with class ${styles.chip} on the ${chipType} container element`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
+      </Chip>
+    );
+
+    expect(screen.getByTestId('container')).toHaveClass(styles.chip);
+  });
+
+  test(`Renders with class ${styles.chipContent} around the ${chipType} content`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').firstChild).toHaveClass(styles.chipContent);
+  });
+
+  test(`Renders ${chipType} children with class ${styles.chipText}`, () => {
+    render(<Chip isOverflowChip={isOverflowChip}>Chip text</Chip>);
+
+    expect(screen.getByText('Chip text')).toHaveClass(styles.chipText);
+  });
+
+  test(`Renders with custom class on the ${chipType} container element`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container" className="custom-class">
+        Chip text
+      </Chip>
+    );
+
+    expect(screen.getByTestId('container')).toHaveClass('custom-class');
   });
 
-  test('closable', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" id="chip_one">
-        Chip
+  test(`Renders a badge when badge prop is passed to ${chipType}`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} badge={<div>Badge content</div>}>
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByText('Badge content')).toBeVisible();
   });
 
-  test('closable with tooltip', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" id="chip_one">
-        1234567890123456789
+  test(`Renders with div container on ${chipType} by default`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').tagName).toBe('DIV');
   });
 
-  test('readonly', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" isReadOnly>
-        4 more
+  test(`Renders with custom container on ${chipType} when component prop is passed`, () => {
+    render(
+      <Chip component="button" isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').tagName).toBe('BUTTON');
   });
```

Comment:
So the reason I did this I think was because a default chip renders a close button so there's multiple buttons. I'm just updating this test so that `component="li"` and querying the listitem role